examples/librispeech/s0/run.sh

```diff
@@ -183,27 +183,37 @@ if [ ${stage} -le 5 ] && [ ${stop_stage} -ge 5 ]; then
     # -1 for full chunk
     decoding_chunk_size=
     ctc_weight=0.5
+    # Polling GPU id
+    num_gpus=$(echo $CUDA_VISIBLE_DEVICES | awk -F "," '{print NF}')
+    gpu_idx=0
     for test in $recog_set; do
     for mode in ${decode_modes}; do
     {
-        test_dir=$dir/${test}_${mode}
-        mkdir -p $test_dir
-        python wenet/bin/recognize.py --gpu -1 \
-            --mode $mode \
-            --config $dir/train.yaml \
-            --test_data $wave_data/$test/format.data \
-            --checkpoint $decode_checkpoint \
-            --beam_size 10 \
-            --batch_size 1 \
-            --penalty 0.0 \
-            --dict $dict \
-            --result_file $test_dir/text_bpe \
-            --ctc_weight $ctc_weight \
-            ${decoding_chunk_size:+--decoding_chunk_size $decoding_chunk_size}
-        tools/spm_decode --model=${bpemodel}.model --input_format=piece < $test_dir/text_bpe | sed -e "s/▁/ /g" > $test_dir/text
-        python tools/compute-wer.py --char=1 --v=1 \
-            $wave_data/$test/text $test_dir/text > $test_dir/wer
-    } &
+        {
+            test_dir=$dir/${test}_${mode}
+            mkdir -p $test_dir
+            python wenet/bin/recognize.py --gpu $gpu_idx \
+                --mode $mode \
+                --config $dir/train.yaml \
+                --test_data $wave_data/$test/format.data \
+                --checkpoint $decode_checkpoint \
+                --beam_size 10 \
+                --batch_size 1 \
+                --penalty 0.0 \
+                --dict $dict \
+                --result_file $test_dir/text_bpe \
+                --ctc_weight $ctc_weight \
+                ${decoding_chunk_size:+--decoding_chunk_size $decoding_chunk_size}
+            tools/spm_decode --model=${bpemodel}.model --input_format=piece < $test_dir/text_bpe | sed -e "s/▁/ /g" > $test_dir/text
+            python tools/compute-wer.py --char=1 --v=1 \
+                $wave_data/$test/text $test_dir/text > $test_dir/wer
+        } &
+
+        ((gpu_idx+=1))
```

Comment:
There is strong assumption that the gpu id is continuous,  please refer to the training code above to polling GPU